test/models/friendship_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'test_helper'
+
+class FriendshipTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+  def setup
+  	@friendship = friendships(:one)
+  	@self_friendship = friendships(:self_friendship)
+    @user = users(:three)
+    @friend = users(:four)
+  end	
+  test 'Valid friendship' do
+	  assert @friendship.valid?  
+  end
+
+  test 'Invalid friendship' do
+  	refute @self_friendship.valid?
+  end
+
+  test 'User has friend after friendship created' do
```

Comment:
In relation to #4 let us also then test that only one friendship is created but is reflected to both sides.